.github/workflows/ci.yml

```diff
@@ -65,14 +62,14 @@ jobs:
                                ./gradlew publishToMavenLocal"
           
       - name: Upload Coverage Report
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@v1
```

Comment:
why do we change the version?

Reply:
Our recent CI fails the task of upload coverage report ([ref](https://github.com/opensearch-project/skills/actions/runs/7663590574/job/20886629954)).
Based on the error log:
```
/usr/bin/docker exec  0dc3635fb865fc78d3ab1313d71902228fb8a54822cfa2e7b9cc6ff78f1d73ad sh -c "cat /etc/*release | grep ^ID"
/__e/node20/bin/node: /lib64/libm.so.6: version `GLIBC_2.2[7](https://github.com/opensearch-project/skills/actions/runs/7663590574/job/20886629954#step:6:8)' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libc.so.6: version `GLIBC_2.2[8](https://github.com/opensearch-project/skills/actions/runs/7663590574/job/20886629954#step:6:9)' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libc.so.6: version `GLIBC_2.25' not found (required by /__e/node20/bin/node)
```
The codecov/codecov-action@v3 use node20 dependency, while we don't have it in linux container. So I changed this to codecov/codecov-action@v1, this is consistent with other plugins.